orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -293,27 +304,24 @@ def subscription_instance_in_use_by(
     if filter_statuses:
         in_use_by_instances = [sub for sub in in_use_by_instances if sub.subscription.status in filter_statuses]
 
-    return list(
-        filter(
-            lambda sub_id: sub_id != subscription_instance.subscription_id,
-            {sub.subscription_id for sub in in_use_by_instances},
-        )
-    )
+    unique_ids = {sub.subscription_id for sub in in_use_by_instances}
```

Comment:
Small thought/remark (optional since it's existing code)

This endpoint needs to return subscription ID's, but to do that it instantiates the entire SubscriptionTable model, even including duplicates.

Perhaps we can at least eliminate duplicates by using [`.unique()`](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.unique)? Like `subscription_instance.in_use_by.unique()` (might be the wrong syntax)

Reply:
Hm. It already does a join with calling `subscription_instance.in_use_by`.
It's clearer to understand this way (less coupled to the db schema) and might be fast enough because of the existing filtering.

I suggest to revisit this as a separate issue if performance problems arise